src/NUnitFramework/tests/Internal/TestExecutionContextTests.cs

```diff
@@ -907,6 +918,65 @@ public void SetAndRestoreValueFormatter()
 
         #endregion
 
+        #region AddTestAttachment
+
+        [Test]
+        public void FilePathOnlyAddedToTestAttachments()
+        {
+            var context = CreateFakeContext();
+            context.AddTestAttachment(_tempFilePath);
+
+            Assert.That(context.CurrentResult.TestAttachments, Has.Exactly(1).Property("Filepath").EqualTo(_tempFilePath).And.Property("Description").Null);
+        }
+
+        [Test]
+        public void FilePathAndDescriptionAddedToTestAttachments()
+        {
+            var context = CreateFakeContext();
+            context.AddTestAttachment(_tempFilePath, "Description");
+
+            Assert.That(context.CurrentResult.TestAttachments, Has.Exactly(1).Property("Filepath").EqualTo(_tempFilePath).And.Property("Description").EqualTo("Description"));
+        }
+
+        [Test]
+        public void RelativePathsMadeAbsolute()
+        {
+            var context = CreateFakeContext();
+            context.AddTestAttachment(TempFileName);
+
+            Assert.That(context.CurrentResult.TestAttachments, Has.Exactly(1).Property("Filepath").EqualTo(Path.Combine(TestContext.CurrentContext.WorkDirectory, TempFileName)));
+        }
+
+        [TestCase(null)]
+#if !PORTABLE && !NETSTANDARD1_6
+        [TestCase("bad<>path.png", IncludePlatform = "Windows")]
+#endif
+        public void InvalidFilePathsThrowsArgumentException(string filepath)
+        {
+            var context = CreateFakeContext();
+            Assert.That(() => context.AddTestAttachment(filepath), Throws.InstanceOf<ArgumentException>());
+        }
+
+#if !PORTABLE
+        [Test]
+        public void NoneExistandFileThrowsFileNotFoundException()
+        {
+            var context = CreateFakeContext();
+            Assert.That(() => context.AddTestAttachment("NotAFile.txt"), Throws.InstanceOf<FileNotFoundException>());
+        }
+
+        public TestExecutionContext CreateFakeContext()
+        {
+            var fakeResult = TestUtilities.Fakes.GetTestMethod(this, "FakeMethod").MakeTestResult();
+            var context = new TestExecutionContext { CurrentResult = fakeResult };
+            return context;
+        }
+
```

Comment:
`CreateFakeContext()` is pretty ugly. Am I missing a better way to create temporary context to test with? I didn't want to attach anything to the actual TestExecutionContext.